CoffeeShopApp/src/model/Order.java

```diff
@@ -0,0 +1,61 @@
+
+
+package model;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class Order {
+    private Map<Product, Integer> items;
+
+    public Order() {
+        this.items = new HashMap();
+    }
+
+    public Order(List<Coffee> coffees, List<Treat> treats) {
+        this();
+        Iterator var3 = coffees.iterator();
+
+        while(var3.hasNext()) {
+            Coffee coffee = (Coffee)var3.next();
+            this.addItem(coffee, 1);
+        }
+
+        var3 = treats.iterator();
+
+        while(var3.hasNext()) {
+            Treat treat = (Treat)var3.next();
+            this.addItem(treat, 1);
+        }
+
+    }
+
+    public void addItem(Product product, int quantity) {
+        this.items.put(product, (Integer)this.items.getOrDefault(product, 0) + quantity);
+    }
+
+    public Map<Product, Integer> getItems() {
+        return this.items;
+    }
+
+    public double getTotalPrice() {
+        return this.items.entrySet().stream().mapToDouble((entry) -> {
+            return ((Product)entry.getKey()).getPrice() * (double)(Integer)entry.getValue();
+        }).sum();
```

Comment:
return this.items.entrySet().stream()
			.mapToDouble(entry -> entry.getKey().getPrice() * entry.getValue())
			.sum();

or

		double totalSum = 0;
		for (var entrySet : items.entrySet()) {
			var quantity = entrySet.getValue();
			double sum = entrySet.getKey().getPrice() * quantity;
			totalSum += sum;
		}